crypto/evp/evp_lib.c

```diff
@@ -272,16 +272,38 @@ int EVP_CIPHER_type(const EVP_CIPHER *ctx)
     }
 }
 
-int EVP_CIPHER_block_size(const EVP_CIPHER *cipher)
+int evp_cipher_cache_constants(EVP_CIPHER *cipher)
 {
     int ok;
-    size_t v = cipher->block_size;
-    OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
-
-    params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_BLOCK_SIZE, &v);
+    size_t ivlen = 0;
+    size_t blksz = 0;
+    size_t keylen = 0;
+    unsigned int mode = 0;
+    unsigned long flags = 0;
+    OSSL_PARAM params[6];
+
+    params[0] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_BLOCK_SIZE, &blksz);
+    params[1] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_IVLEN, &ivlen);
+    params[2] = OSSL_PARAM_construct_size_t(OSSL_CIPHER_PARAM_KEYLEN, &keylen);
+    params[3] = OSSL_PARAM_construct_uint(OSSL_CIPHER_PARAM_MODE, &mode);
+    params[4] = OSSL_PARAM_construct_ulong(OSSL_CIPHER_PARAM_FLAGS, &flags);
+    params[5] = OSSL_PARAM_construct_end();
     ok = evp_do_ciph_getparams(cipher, params);
+    if (ok) {
+        /* Provided implementations may have a custom cipher_cipher */
+        if (cipher->prov != NULL && cipher->ccipher != NULL)
+            flags |= EVP_CIPH_FLAG_CUSTOM_CIPHER;
```

Comment:
Why is this necessary.  `EVP_CIPH_FLAG_CUSTOM_CIPHER` is only looked at in legacy code, and the place where we do use `ccipher`, we check for its existance, not this flag:

https://github.com/openssl/openssl/blob/bdbf2df2e685ae653f3c683ce2f734eb0c0888e0/crypto/evp/evp_lib.c#L312-L316

Reply:
Happy to take it out,
It was added in
https://github.com/openssl/openssl/commit/f7397f0d58ce7ddf4c5366cd1846f16b341fbe43
The documentation for EVP_cipher that was changed in this commit seems to not be correct now.. It said it looks for the flag (it doesnt seem to though??)

Reply:
"historical semantics"...  oh, right, that.

No, *we* don't check because we don't need to.  However, callers may want to know what semantics to expect, and the only recourse *they* have is that flag.

[runs in circles, screaming in horror]

Reply:
So yeah, let's leave this alone.